### PR TITLE
Ar 791 my library

### PR DESCRIPTION
**Pull request by davgur**

### CODE REVIEW

File: public/locales/en/common.json [hunk 1]
```diff
@@ -114,7 +115,9 @@
     "previous": "Previous",
     "next": "Next",
     "copy": "Copy",
-    "view-all": "View All"
+    "view-all": "View All",
```
Comment (gshilin) [inline public/locales/en/common.json:118]: Please make translations to all languages the same.

File: public/locales/en/common.json [hunk 2]
```diff
@@ -400,7 +403,10 @@
       "info": {
         "sources": "Kabbalah Sources",
         "topics": "Topics",
-        "collections": "Related to"
+        "collections": "Related to",
```
Comment (gshilin) [inline public/locales/en/common.json:406]: Please make translations to all languages the same.

File: public/locales/en/common.json [hunk 3]
```diff
@@ -623,5 +629,35 @@
     "copy-link": "Copy link",
     "copy-text": "Copy text",
     "disable-share": "Cancel"
+  },
+  "personal": {
```
Comment (gshilin) [inline public/locales/en/common.json:633]: Please make translations to all languages the same.

File: public/locales/ru/common.json [hunk 4]
```diff
@@ -726,5 +732,35 @@
     "copy-link": "Копировать ссылку",
     "copy-text": "Копировать текст",
     "disable-share": "Отменить"
+  },
+  "personal": {
+    "subsNewUnits": "новых эпизодов",
+    "unsubscribe": "Вы подписаны",
```
Comment (gshilin) [inline public/locales/ru/common.json:738]: Are you sure?

File: src/components/Layout/Login.js [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+const Login = ({ t, language }) => {
+  const [isActive, setIsActive] = useState(false);
+  const { isMobileDevice }      = useContext(DeviceInfoContext);
+  const direction               = getLanguageDirection(language);
+  const popupStyle              = { direction };
+  const dispatch                = useDispatch();
+  const user                    = useSelector(state => selectors.getUser(state.auth));
+
+  useEffect(() => {
+    !user && initKC(dispatch, language);
```
Comment (gshilin) [inline src/components/Layout/Login.js:22]: what happens when you change language before logging in?

File: src/components/Pages/Unit/widgets/Info/Info.js [hunk 6]
```diff
@@ -53,11 +61,15 @@ const makeTagLinks = (tags = [], getTagById) =>
...
+      return <Link key={id} to={`/topics/${id}`}>
+        <Button basic size="tiny" className="link_to_cu">
+          {label}
+        </Button>
+      </Link>;
+    }), ''));
 
 const makeSourcesLinks = (sources = [], getSourceById, filteredListPath) => Array.from(intersperse(
-  sources.map(x => {
+  sources.map((x, i) => {
```
Comment (gshilin) [inline src/components/Pages/Unit/widgets/Info/Info.js:72]: Why did you add `i` here? It's not used...

File: src/components/Pages/Unit/widgets/Info/Info.js [hunk 7]
```diff
@@ -114,70 +127,117 @@ const makeCollectionsLinks = (collections = {}, t, currentCollection) => {
...
+  return t(tName, { name: episode });
+};
+
+const getEpisodeInfo = (ct, cIDs, currentCollection, filmDate, t) => {
+  const cIds        = cIDs && Object.keys(cIDs);
+  const cId         = cIds && (currentCollection ? cIds.find(c => c.split('_')[0] === currentCollection.id) : cIds[0]);
+  const showEpisode = cId && cId.indexOf(CT_KTAIM_NIVCHARIM) === -1;
+  const episode     = showEpisode && cId.split('_').slice(-1).pop();
+  const episodeInfo = [];
+  if (episode && episode !== '0')
```
Comment (gshilin) [inline src/components/Pages/Unit/widgets/Info/Info.js:141]: add curly braces

File: src/components/Pages/Unit/widgets/Info/Info.js [hunk 8]
```diff
@@ -114,70 +127,117 @@ const makeCollectionsLinks = (collections = {}, t, currentCollection) => {
...
+    episodeInfo.push(getEpisodeName(ct, episode, t));
+  episodeInfo.push(t('values.date', { date: filmDate }));
+  const len = episodeInfo.length - 1;
+  return episodeInfo.map((x, i) => (
+    <span key={i}>
+      {x}
+      {i < len && (<span className="seperator">|</span>)}
```
Comment (gshilin) [inline src/components/Pages/Unit/widgets/Info/Info.js:148]: `separator`

File: src/helpers/Api.js [hunk 9]
```diff
@@ -7,6 +8,7 @@ const IMAGINARY_URL           = process.env.REACT_APP_IMAGINARY_URL;
 const IMAGINARY_INTERNAL_HOST = process.env.REACT_APP_IMAGINARY_INTERNAL_HOST || 'localhost';
 const API_FEED                = process.env.REACT_APP_FEED;
 const CHRONICLES_BACKEND      = process.env.REACT_APP_CHRONICLES_BACKEND;
+const MY_BACKEND              = process.env.REACT_APP_MY_BACKEND;
```
Comment (gshilin) [inline src/helpers/Api.js:11]: Any chance to name it more clear?

File: src/helpers/player.js [hunk 10]
```diff
@@ -185,6 +185,18 @@ const playlist = (collection, mediaType, contentLanguage, uiLanguage) => {
   };
 };
 
+const playlistFromUnits = (collection, mediaType, contentLanguage, uiLanguage) => {
+  const items = collection.content_units
+    .map(x => playableItem(x, mediaType, uiLanguage, contentLanguage))
+    .filter(item => !!item.unit)
```
Comment (gshilin) [inline src/helpers/player.js:191]: You can check `item.unit` inside of `playableItem()` and eliminate second `.map()`.
Anyway it is better to `.filter` first and to `.map()` shorter lists.

File: src/sagas/helpers/keycklockManager.js [hunk 11]
```diff
@@ -0,0 +1,84 @@
...
+//for SSR use empty functions
+let login  = () => console.error('Must be override on browser, keycloak');
+let logout = () => console.error('Must be override on browser, keycloak');
+
+//because of keycloak-js module use windows on init we import it dynamic
+let keycloak;
+
+//demi authentication
+/*
+export const initKC_ = (dispatch) => {
```
Comment (gshilin) [inline src/sagas/helpers/keycklockManager.js:12]: Can we remove this commented code?

File: src/stylesheets/includes/_video-player.scss [hunk 12]
```diff
@@ -7,9 +7,11 @@
     background-color: $darkWhite;
     width: 100% !important;
 
-    #unit_container {
+    .unit_container {
       @include mobile {
-        padding: 0;
+        padding-bottom: 0;
```
Comment (gshilin) [inline src/stylesheets/includes/_video-player.scss:12]: padding: auto 0 0;

**Review by gshilin — CHANGES_REQUESTED** [review]

File: public/locales/ru/common.json (hunk 4 above)
Comment (davgur) [reply public/locales/ru/common.json:738]: button with action unsubscribe where wright that you subscribed

File: src/helpers/player.js (hunk 10 above)
Comment (davgur) [reply src/helpers/player.js:191]: Was think that stream functions like map/filter ... are use one interaction. So its not affect on performance ?

File: src/stylesheets/includes/_video-player.scss (hunk 12 above)
Comment (davgur) [reply src/stylesheets/includes/_video-player.scss:12]: image

File: src/components/Pages/Unit/widgets/Info/Info.js (hunk 8 above)
Comment (davgur) [reply src/components/Pages/Unit/widgets/Info/Info.js]: fixed

File: src/helpers/Api.js (hunk 9 above)
Comment (davgur) [reply src/helpers/Api.js]: changed

File: src/sagas/helpers/keycklockManager.js (hunk 11 above)
Comment (davgur) [reply src/sagas/helpers/keycklockManager.js]: removed

File: public/locales/en/common.json (hunk 2 above)
Comment (davgur) [reply public/locales/en/common.json:406]: added

File: public/locales/en/common.json (hunk 3 above)
Comment (davgur) [reply public/locales/en/common.json:633]: added

File: src/components/Layout/Login.js (hunk 5 above)
Comment (davgur) [reply src/components/Layout/Login.js:22]: must to call init on first render only

File: src/helpers/player.js (hunk 10 above)
Comment (gshilin) [reply src/helpers/player.js:191]: That's not true, but who cares

**Review by gshilin — APPROVED** [review]
